cli/Valet/Brew.php

```diff
@@ -370,21 +370,59 @@ function ($exitCode, $errorOutput) use ($formula) {
         );
     }
 
+    /**
+     * Get all the currently running brew services.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function getAllRunningServices()
+    {
+        return $this->getRunningServicesAsRoot()
+            ->concat($this->getRunningServicesAsUser())
+            ->unique();
+    }
+
+    /**
+     * Get the currently running brew services as root.
+     * i.e. /Library/LaunchDaemons (started at boot).
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function getRunningServicesAsRoot()
+    {
+        return $this->getRunningServices();
+    }
+
     /**
      * Get the currently running brew services.
+     * i.e. ~/Library/LaunchAgents (started at login).
      *
      * @return \Illuminate\Support\Collection
      */
-    public function getRunningServices()
+    public function getRunningServicesAsUser()
     {
-        return collect(array_filter(explode(PHP_EOL, $this->cli->runAsUser(
-            'brew services list | grep started | awk \'{ print $1; }\'',
-            function ($exitCode, $errorOutput) {
-                output($errorOutput);
+        return $this->getRunningServices(true);
+    }
 
-                throw new DomainException('Brew was unable to check which services are running.');
-            }
-        ))));
+    /**
+     * Get the currently running brew services.
+     *
+     * @param  bool  $asUser
+     * @return \Illuminate\Support\Collection
+     */
+    public function getRunningServices($asUser = false)
```

Comment:
My only concern with this PR is this syntax--I'm not a big fan of boolean parameters. I'm trying to think about how we can rework this to instead be a syntax that's clearer when you're calling it. I'll keep thinking, but other than that this PR looks great--thank you for making it!

Reply:
Here's an idea from @driftingly:

```php
    public function getCommandLineArgs($prefix = ''): array
    {
        $command = $prefix.'brew services list | grep started | awk \'{ print $1; }\'';
        $onError = function ($exitCode, $errorOutput) {
            output($errorOutput);

            throw new DomainException('Brew was unable to check which services are running.');
        };

        return [$command, $onError];
    }

    public function getRunningServicesAsRoot(): Collection
    {    
        return $this->formatCommandLineOutput($this->cli->run(...$this->getCommandLineArgs('sudo ')));
    }

    public function getRunningServicesAsUser(): Collection
    {
        return $this->formatCommandLineOutput($this->cli->runAsUser(...$this->getCommandLineArgs()));
    }

    public function formatCommandLineOutput($output): Collection
    {
        return collect(array_filter(explode(PHP_EOL, $output)));
    }
```

(I added the `prefix` concept because it was missing from his refactor). What do you think?

Reply:
Hey @mattstauffer, sorry for the delay on this, I somehow missed the notifications on this. 

I actually tried a few approaches to avoid the boolean param but in the end, decided it was not worth it. I actually went as far as creating a new class that would actually just be a wrapper for the list command, that would return the output as a collection that could then be filtered by started, stopped, error, etc. But decided against it as maybe adding a new class for only 3-4 lines of code didn't feel right. What do you think?

For the solution above, I actually tried a similar approach but decided against it as I couldn't find any good name for the functions. `getCommandLineArgs` and `formatCommandLineOutput` feels a bit too general? i.e. `Brew@getCommandLineArgs($prefix)` does not offer any info about the fact that this is actually only about the list command and it only gets "started" services? So I'd prefer having the boolean param over needing to add these 2 new functions without a proper name. But honestly, both solutions are in the "not ideal" column for me so I'm fine with both. So if you guys think this approach is better, I'll make the update, should be an easy one. Let me know :)